DeviceAdministration/Infrastructure/BusinessLogic/DeviceLogic.cs

```diff
@@ -384,29 +381,15 @@ await _virtualDeviceStorage.AddOrUpdateDeviceAsync(new InitialDeviceConfig()
             {
                 throw new DeviceRequiredPropertyNotFoundException("Required DeviceProperties not found");
             }
-
-            if ((dynamicMetaObjectProvider = device.DeviceProperties as IDynamicMetaObjectProvider) != null)
```

Comment:
yeap it's running fine. The code that I've deleted was hit when device was dynamic...